test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/Common/BaseE2E.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Common
+{
+	/* This class acts as the base class for all the language test case classes.
+	* Takes care of Initial Orchestration and actual flow of the test.
+	*/
+	public class BaseE2E
+	{
+		private readonly KafkaE2EFixture _kafkaE2EFixture;
+		private readonly Language _language;
+		private readonly BrokerType _brokerType;
+		private readonly E2ETestInvoker _invoker;
+		private readonly ILogger _logger = TestLogger.GetTestLogger();
+
+		protected BaseE2E(KafkaE2EFixture kafkaE2EFixture, Language language, BrokerType brokerType, ITestOutputHelper output)
+		{
+			_kafkaE2EFixture = kafkaE2EFixture;
```

Comment:
Generally the infra setup is the responsibility of the test case in a setup() phase. The test case when it is run should not worry about infra setup. Is there a setup phase in xunit which you can utilize ? In that way the constructor which is an E2E test case can focus on only doing the minimum stuff (light weight) which is required for running a specific test case. We can discuss offline for more details.

Reply:
In xunit this responsibility lies with the fixture. In the current implementation the test cases do not take care about the infra setup rather for each test case xunit calls the fixture which to take care of the setup and the disposal phases.